src/processor/operator/recursive_extend/bfs_scheduler.cpp

```diff
@@ -0,0 +1,192 @@
+#include "processor/operator/recursive_extend/bfs_scheduler.h"
+
+#include "chrono"
+
+namespace kuzu {
+namespace processor {
+
+/**
+ * Main function of the scheduler that dispatches morsels from the FTable between threads. Or it can
+ * be called if a thread could not find a morsel from it's local SSSPSharedState.
+ *
+ * @return - TRUE indicates check the combination value of the state pair, FALSE indicates some
+ * work was found hence the state's values can be ignored.
+ */
+bool MorselDispatcher::getBFSMorsel(
+    const std::shared_ptr<FTableSharedState>& inputFTableSharedState,
+    std::vector<common::ValueVector*> vectorsToScan, std::vector<ft_col_idx_t> colIndicesToScan,
+    common::ValueVector* srcNodeIDVector, std::unique_ptr<BaseBFSMorsel>& bfsMorsel,
+    std::pair<GlobalSSSPState, SSSPLocalState>& state) {
+    std::unique_lock lck{mutex};
+    switch (schedulerType) {
+    case SchedulerType::OneThreadOneMorsel: {
+        auto inputFTableMorsel = inputFTableSharedState->getMorsel();
+        if (inputFTableMorsel->numTuples == 0) {
+            state = {COMPLETE, NO_WORK_TO_SHARE};
+            return true;
+        }
+        inputFTableSharedState->getTable()->scan(vectorsToScan, inputFTableMorsel->startTupleIdx,
+            inputFTableMorsel->numTuples, colIndicesToScan);
+        bfsMorsel->resetState();
+        auto nodeID = srcNodeIDVector->getValue<common::nodeID_t>(
+            srcNodeIDVector->state->selVector->selectedPositions[0]);
+        bfsMorsel->markSrc(nodeID);
+        return false;
+    }
+    case SchedulerType::nThreadkMorsel: {
+        switch (globalState) {
+        case COMPLETE: {
+            state = {globalState, NO_WORK_TO_SHARE};
```

Comment:
use COMPLETE directly instead of globalState. Apply the same style across this PR

Reply:
done